input.go

```diff
@@ -0,0 +1,19 @@
+package shaping
+
+import (
+	"github.com/go-text/di"
+	"github.com/go-text/font"
+	"golang.org/x/image/math/fixed"
+)
+
+type Input interface {
+	// Text returns the characters to be shaped.
+	Text() []rune
+	// Direction returns the directionality of the text.
+	Direction() di.Direction
+	// Face returns the font face to render the text in.
+	Face() font.Face
```

Comment:
I would propose a slight variation on this method, essentially as an optimization. The issue is that the shaper will have to do some additional processing on the Face input (like loading layout tables). This work only needs  to be executed once, whereas with the current interface, each call to Shape would trigger this step.

One way to avoid this performance hit would be to cache the information for each `Face`, in the `Shaper` implementation.

I would suggest an alternative, which is what Harfbuzz does : expose a new 
```
type Font struct {
    face font.Face
    // additional private fields computed from face 

    Size fixed.Int26_6  
}
```

and a method `NewFont(face font.Face) *Font`, and replace the current methods
```
Face() font.Face 
Size() fixed.Int26_6
```
of the `Input` interface with the single `Font() *font.Font`.
In a way, this just defers the caching to the client code, but it could be more natural.

Also note that a wrapper object like `Font` is needed anyway to handle the scaling of the position values given by the `Face`.